File: 0118_Pascal's Triangle/Pascal'sTriangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> ans = new ArrayList<List<Integer>>();
        ans.add(new ArrayList<Integer>());
        ans.get(0).add(1);
            
        for(int i=1;i<numRows;i++) {
            ans.add(new ArrayList<Integer>());
            List<Integer> prevList = ans.get(i-1);
            List<Integer> currList = ans.get(i);
            int len = prevList.size() + 1;
            for(int j=0;j<len;j++)
            {
                if(j == 0) {
                    currList.add(prevList.get(j));
                }else if(j == len - 1) {
                    currList.add(prevList.get(j-1));
                }else{
                    currList.add(prevList.get(j-1) + prevList.get(j));
                }
            }
        }
        
        return ans;
    }
}

File: 0118_Pascal's Triangle/Pascal'sTriangle.cs
public class Solution {
    public IList<IList<int>> Generate(int numRows) {
        var ans = new List<IList<int>>();
        
        var list = new List<int>();
        list.Add(1);
        ans.Add(list);
        for(int i=1;i<numRows;i++)
        {
            if(ans.Count == i)
                ans.Add(new List<int>());
            
            var prevList = ans[i-1];
            var currList = ans[i];
            
            var count = prevList.Count + 1;
            for(int j=0;j< count ;j++)
            {
                if(j == 0){
                    currList.Add(prevList[j]);
                }else if(j+1 == count) {
                    currList.Add(prevList[j-1]);
                }else {
                    currList.Add(prevList[j-1] + prevList[j]);
                }
            }
        }
        
        return ans;
    }
}